### DISCUSSION TITLE
shiny's 'execution halted' message

### DISCUSSION THREAD

**Discussion by road2you** [quarto-dev/quarto-cli#984] [Q&A] [answered]
I can't execute the `Hello, Shiny` example. 
The error message is as below.

```
Error in !is.na(preserve) || isTRUE(format$render$`code-link`) : 
  'length = 2' in coercion to 'logical(1)'
Calls: .main -> execute
Execution halted
```
The separated version that creates a `server.R` file separately works like a charm.

Following is my session info :
```
R Under development (unstable) (2022-05-25 r82401 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Korean_Korea.utf8  LC_CTYPE=Korean_Korea.utf8    LC_MONETARY=Korean_Korea.utf8
[4] LC_NUMERIC=C                  LC_TIME=Korean_Korea.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] digest_0.6.29    R6_2.5.1         fastmap_1.1.0    xfun_0.31        knitr_1.39       htmltools_0.5.2 
 [7] rmarkdown_2.14   cli_3.3.0        askpass_1.1      openssl_2.0.2    sass_0.4.1       jquerylib_0.1.4 
[13] rsconnect_0.8.25 compiler_4.3.0   tools_4.3.0      curl_4.3.2       evaluate_0.15    bslib_0.3.1     
[19] yaml_2.3.5       rlang_1.0.2      jsonlite_1.8.0  
```
quarto version: `0.9.462`
RStudio version: `2022.02.2 Build 485`

I did not make any modifications to the simple example code.

**Comment by road2you** [comment]
All three shiny examples(`Old Faithful`, `K-Means`, `Diamonds`) can't be executed. 

For `Old Faithful` and `K-Means`, rendering works by creating a separate `server.R` file. 
For `Diamonds`, by separating `server.R` file **AND** removing the chunk argument of `#| context: setup` (and add `library(ggplot2)` to `server.R`) rendering works. 

I think it may be that the chunk argument related to shiny is not properly handled.

**Comment by jjallaire** [comment, marked as answer]
I am seeing a warning rather than an error (perhaps in your config warnings are treated as errors?)

In any case, I have fixed the code to eliminate the warning here (https://github.com/quarto-dev/quarto-cli/commit/b7878d22688aa60c243862ddbf77805d36dc8f34) so this should no longer be an issue.

**Reply by road2you** [reply]
After installing quarto version 0.9.476, the rendering proceeds to the end and succeeds. No error message appears. Thank you.
